### PR TITLE
fix(`WebGLProbe`): regression `enableGLFiltering` config + init bug

### PR DESCRIPTION
**Pull request by ShaMan123**
This PR fixes 2 regression from #8199 
- `enableWebGLFiltering` see https://github.com/fabricjs/fabric.js/pull/8199/files#r973981355
- bug in negation

and adds tests

### CODE REVIEW

File: src/filters/WebGLProbe.ts [hunk 1]
```diff
@@ -57,10 +58,10 @@ class WebGLProbe {
    * @returns config object if true
    */
   private queryWebGL() {
-    if (this.initialized || !fabric.isLikelyNode) {
+    if (this.initialized || fabric.isLikelyNode) {
```
Comment (asturur) [inline src/filters/WebGLProbe.ts:61]: mmmm when this changed?

**Review by asturur — APPROVED** [review]

**Comment by asturur** [comment]
Add the changelog please!

File: src/filters/WebGLProbe.ts (hunk 1 above)
Comment (ShaMan123) [inline src/filters/WebGLProbe.ts:61]: 😵
😲
regression found

**Review by ShaMan123 — COMMENTED** [review]
fixed regressions from #8199 
- `enableWebGLFiltering`
- bug in negation
- added tests

## To Do
Once my ci PRs are in I can test the WebGLProbe with sinon
In a next PR we should make config changes restored after each test

File: src/filters/WebGLProbe.ts (hunk 1 above)
Comment (ShaMan123) [reply src/filters/WebGLProbe.ts:61]: a typo

**Comment by asturur** [comment]
what do you mean `Once my ci PRs are in I can test the WebGLProbe with sinon` ? We can test webGlProbe right now probably launching a browser with acceleration disabled.

I'm not familiar with sinon anymore, but i would avoid adding more testing frameworks right now. if i have to differentiate testing i want to put down all the choices and pick reasonably. ( node specific and browser specific )

**Comment by ShaMan123** [comment]
sinon is good for both... it is meant for stubbing and spying after objects, classes, modules etc.
Without it I can't access the webGLProbe because it isn't and shouldn't be exported by fabric

**Comment by ShaMan123** [comment]
I remember adding sinon to a ci pr

**Review by ShaMan123 — COMMENTED** [review]
added changelog

**Comment by asturur** [comment]
i used sinon a lot, i don't use it anymore, i m not sure i want to re-learn it.
Adding Sinon and committing to it is something that needs to be weighted against moving to jest for unit testing and something else for browser-needed testing.

> I remember adding sinon to a ci pr

Well is not in master yet.

**Review by asturur — APPROVED** [review]

**Pull request merged**